rbc/external.py

```diff
@@ -44,27 +47,28 @@ def __init__(self, name: str, signature: types.FunctionType):
 
     def register(self):
         # typing
-        from numba.core.typing.templates import (
-            make_concrete_template,
-            infer_global,
-            infer,
-        )
+        class ExternalTemplate(typing.templates.AbstractTemplate):
+            obj = self
+            key = self.name
 
-        template = make_concrete_template(
-            self.name, key=self.name, signatures=[self.signature]
-        )
-        infer(template)
-        infer_global(self, types.Function(template))
+            def generic(self, args, kws):
+                t = Type.fromobject(self.obj._signature).tonumba()
```

Comment:
I presume that in multiple signature cases (see below), this is the place where `t` needs to depend on `args`.

Reply:
exactly